app/src/main/java/com/codingblocks/cbonlineapp/auth/onboarding/OnBoardingActivity.kt

```diff
@@ -35,30 +31,60 @@ class OnBoardingActivity : BaseCBActivity() {
         }
     }
 
-    private fun setAdapter() {
-        viewPager.apply {
-            adapter = mAdapter
+    private fun setupIntroSlider() {
+        setUpViewPagerWithDots(getIntroList())
+    }
+
+    private fun setUpViewPagerWithDots(introList: List<Intro>) {
+        val dotsCount = introList.size
+        val dots = Array(dotsCount) {
+            ImageView(this)
+        }
+
+        initDots(dots)
+        viewPager2.apply {
+            adapter = IntroPagerAdapter(introList)
             currentItem = 0
             offscreenPageLimit = 1
-//            addOnPageChangeListener(
-//                    pageChangeCallback { pos, _, _ ->
-//                        for (i in 0 until dotsCount) {
-//                            dots[i].isSelected = false
-//                        }
-//                        dots[pos].isSelected = true
-//                    }
-//            )
-            setScrollDuration(200)
+            registerOnPageChangeCallback(
+                pageChangeCallback { pos, _, _ ->
+                    for (i in 0 until dotsCount) {
+                        dots[i].isSelected = false
+                    }
+                    dots[pos].isSelected = true
+                })
         }
-        dotsCount = mAdapter.count
-        for (i in 0..2) {
+    }
+
+    private fun initDots(dots: Array<ImageView>) {
+        for (i in dots.indices) {
             dots[i] = ImageView(this)
-            dots[i].setImageDrawable(getDrawable(R.drawable.dots))
+            dots[i].setImageDrawable(ContextCompat.getDrawable(this, R.drawable.dots))
             dots[i].isSelected = false
             val params = LinearLayout.LayoutParams(20, 20, 1f)
             params.setMargins(10, 0, 10, 0)
             viewPagerCountDots.addView(dots[i], params)
         }
         dots[0].isSelected = true
     }
+
+    private fun getIntroList(): List<Intro> {
+        return listOf(
+            Intro(
```

Comment:
is this code fine ? or you have any better approaches @aggarwalpulkit596  ?

Reply:
Make an array of drawable and Strings and use that instead much cleaner code that way

Reply:
@aggarwalpulkit596 didn't get you. 

instead of int you are asking to add drawable there ?

Reply:
https://github.com/coding-blocks/CBOnlineApp/blob/development/app/src/main/res/values/arrays.xml refer this

Reply:
Does it help ?